util/afero.go

```diff
@@ -88,20 +88,30 @@ func stringify(fs afero.Fs, name string, indent []bool) (string, error) {
 // dst has a different set of contents in the same location, the behaviour is
 // currently undefined.
 // TODO: this should be made more rsync like and robust!
-func CopyFs(srcFs, dstFs afero.Fs, src, dst string, force bool) error {
-	if src == "" {
+func CopyFs(srcFs afero.Fs, dstFs afero.Fs, src string, dst string, force bool) error {
+	var srcFsLen int
+
+	if len(src) <= 1 {
+		// root directory
 		src = "/"
+		srcFsLen = 1
+	} else {
+		// todo: clean this up with function that gets parent dir?
+		src = strings.TrimSuffix(src, "/")
+		dirs := strings.Split(src, "/")
+		out := strings.Join(dirs[:len(dirs)-1], "/")
+		srcFsLen = len(out)
 	}
-	if dst == "" {
-		dst = "/"
+	if dst == "" || string(dst[len(dst)-1]) != "/" {
```

Comment:
All this string manipulation is kind of complicated. Can you try with the version of `afero.go` in #423 to see if your tests pass with that version?

We should settle on the simplest solution that works for all our tests (and is logically correct.) If you want to discuss the bug, ping me on IRC, or here.